sympy/functions/elementary/hyperbolic.py

```diff
@@ -287,19 +287,19 @@ def _eval_is_positive(self):
         re, im = arg.as_real_imag()
         im_mod = im % (2*pi)
 
-        if im_mod == 0:
+        if im_mod.is_zero:
             return True
 
-        if re == 0:
+        if re.is_zero:
             if im_mod < pi/2 or im_mod > 3*pi/2:
                 return True
             elif im_mod >= pi/2 or im_mod <= 3*pi/2:
                 return False
 
-        return fuzzy_or([fuzzy_and([fuzzy_bool(Eq(re, 0)),
-                         fuzzy_or([fuzzy_bool(im_mod < pi/2),
-                                   fuzzy_bool(im_mod > 3*pi/2)])]),
-                         fuzzy_bool(Eq(im_mod, 0))])
+        return fuzzy_or([fuzzy_and([Eq(re, 0),
+                         fuzzy_or([im_mod < pi/2,
+                                   im_mod > 3*pi/2])]),
+                         Eq(im_mod, 0)])
```

Comment:
`Eq(re, 0)` should be `re.is_zero` (fuzzy bool) rather than symbolic Boolean.

Reply:
This fuzzy logic statement should be able to replace the conditionals above. I need to go but I'll push a commit here later

Reply:
Will wait for your commit.